src/runner/trainers/base_trainer.py

```diff
@@ -220,10 +221,21 @@ def load(self, path):
         checkpoint = torch.load(path, map_location='cpu')
         self.net.load_state_dict(checkpoint['net'])
         self.optimizer.load_state_dict(checkpoint['optimizer'])
-        if checkpoint['lr_scheduler'] is not None:
+
+        if self.lr_scheduler is not None and checkpoint['lr_scheduler'] is not None:
             self.lr_scheduler.load_state_dict(checkpoint['lr_scheduler'])
```

Comment:
I'm not sure whether it fails when the lr scheduler is different from the one saved in the checkpoint.

Reply:
Yep. As long as the `state_dict()` kwargs can be matched, it will not throw error messages. So do `self.net` and `self.optimizer`. I'm not sure how to prevent it and should we prevent it?